### PR TITLE
Avoid dependency on RNR-initialisation

### PR DESCRIPTION
**Pull request by meriac**
In case the RNR register is initialised to a region outside of the supported region count the core behaviour is undefined by the architecture. To avoid that we specifically enforce region number zero.

@AlessandroA @Patater

### CODE REVIEW

File: core/system/src/mpu/vmpu_armv7m.c [hunk 1]
```diff
@@ -68,8 +68,7 @@
 #define ARMv7M_MPU_ALIGNMENT_MASK (ARMv7M_MPU_ALIGNMENT-1)
 
 /* MPU helper macros */
-#define MPU_RBAR(region,addr)   (((uint32_t)(region))|MPU_RBAR_VALID_Msk|addr)
-#define MPU_RBAR_RNR(addr)     (addr)
+#define MPU_RBAR_RNR(region)      (MPU_RBAR_VALID_Msk | (((region)<<MPU_RBAR_REGION_Pos) & MPU_RBAR_REGION_Msk))
```
Comment (AlessandroA) [inline core/system/src/mpu/vmpu_armv7m.c]: Could you add spaces around `<<`?

File: core/system/src/mpu/vmpu_armv7m.c [hunk 2]
```diff
@@ -629,10 +628,11 @@ void vmpu_arch_init(void)
 {
     /* reset MPU */
     MPU->CTRL = 0;
-    /* check MPU region aligment */
-    MPU->RBAR = MPU_RBAR_ADDR_Msk;
+    /* check MPU region aligment using region 0 */
```
Comment (AlessandroA) [inline core/system/src/mpu/vmpu_armv7m.c]: `Check MPU region alignment using region 0.` (we now use always capitalized, punctuated sentences) (also, mind `aligNment`)

**Comment by Patater** [comment]
If there is a GitHub issue this fixes, mention the issue in the commit message.

File: core/system/src/mpu/vmpu_armv7m.c [hunk 3]
```diff
@@ -629,10 +628,11 @@ void vmpu_arch_init(void)
...
+    MPU->RNR = 0;
```
Comment (Patater) [inline core/system/src/mpu/vmpu_armv7m.c]: Do we need to program the RNR to something else later, or is leaving it at 0 okay?

Comment (meriac) [inline core/system/src/mpu/vmpu_armv7m.c]: @Patater leaving it at zero is OK, I wourd rather not want to depend on other code re-initializing this register.

**Comment by Patater** [comment]
LGTM

File: core/system/src/mpu/vmpu_armv7m.c [hunk 4]
```diff
@@ -639,12 +639,14 @@ void vmpu_arch_init(void)
...
     DPRINTF("MPU.REGIONS=%i\r\n", (uint8_t)(MPU->TYPE >> MPU_TYPE_DREGION_Pos));
-    DPRINTF("MPU.ALIGNMENT=0x%08X\r\n", g_vmpu_aligment_mask);
-    DPRINTF("MPU.ALIGNMENT_BITS=%i\r\n", vmpu_bits(g_vmpu_aligment_mask));
+    DPRINTF("MPU.ALIGNMENT=0x%08X\r\n", aligment_mask);
+    DPRINTF("MPU.ALIGNMENT_BITS=%i\r\n", vmpu_bits(aligment_mask));
 
     /* sanity checks */
-    assert(ARMv7M_MPU_REGIONS == ((MPU->TYPE >> MPU_TYPE_DREGION_Pos) & 0xFF));
-    assert(ARMv7M_MPU_ALIGNMENT_BITS == vmpu_bits(g_vmpu_aligment_mask));
+    if(ARMv7M_MPU_REGIONS != ((MPU->TYPE >> MPU_TYPE_DREGION_Pos) & 0xFF))
```
Comment (Patater) [inline core/system/src/mpu/vmpu_armv7m.c]: Please do these assert changes in a separate commit.

**Comment by meriac** [comment]
@AlessandroA @Patater  simplified patch
- removed new MPU_RBAR macro, so `<<`-whitespace-fix is not necessary any more
- split into separate functional patches

**Comment by Patater** [comment]
LGTM

File: core/system/src/mpu/vmpu_armv7m.c [hunk 5]
```diff
@@ -640,13 +640,14 @@ void vmpu_arch_init(void)
...
-    DPRINTF("MPU.ALIGNMENT=0x%08X\r\n", g_vmpu_aligment_mask);
-    DPRINTF("MPU.ALIGNMENT_BITS=%i\r\n", vmpu_bits(g_vmpu_aligment_mask));
+    DPRINTF("MPU.ALIGNMENT=0x%08X\r\n", aligment_mask);
+    DPRINTF("MPU.ALIGNMENT_BITS=%i\r\n", vmpu_bits(aligment_mask));
 
     /* Perform sanity checks. */
-    assert(ARMv7M_MPU_REGIONS == ((MPU->TYPE >> MPU_TYPE_DREGION_Pos) & 0xFF));
-    assert(ARMv7M_MPU_ALIGNMENT_BITS == vmpu_bits(g_vmpu_aligment_mask));
-
+    if(ARMv7M_MPU_REGIONS != ((MPU->TYPE >> MPU_TYPE_DREGION_Pos) & 0xFF))
```
Comment (AlessandroA) [inline core/system/src/mpu/vmpu_armv7m.c]: Could you add a space after the `if` and the [1TBS braces](https://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)?

```C
if (condition) {
    code;
}
```

(also applies to line 649)

**Comment by AlessandroA** [comment]
Added [a comment](https://github.com/ARMmbed/uvisor/pull/194/commits/d255b615b9736061710e4f75860d8fb7617a728c#r56215974). Other than that, LGTM.

**Comment by meriac** [comment]
@AlessandroA added braces to both HALT_ERROR, please merge if you're happy with it.

**Comment by AlessandroA** [comment]
Thanks! LGTM
